prover/src/shared_state.rs

```diff
@@ -360,7 +376,7 @@ impl SharedState {
         let tasks: Vec<ProofRequestOptions> = rw
             .tasks
             .iter()
-            .filter(|&e| e.result.is_none())
+            .filter(|&e| e.result.is_none() && !e.completed)
```

Comment:
```suggestion
            .filter(|&e| e.result.is_none() || !e.completed)
```

Reply:
1. the node gets `sinfo` from other peers, the task status is {result: None, completed: true}, 
=> `e.result.is_none() && !e.completed == true`
2. the node finished the task itself, the task status is {result: Some, completed: true}
=> `e.result.is_none() && !e.completed == false`

So, but the task in situation 2 is also a finished task.

Reply:
This is a tricky point as if itself has the result, the self result is not none because of the edition

Reply:
> BTW:
> {result: None, completed: true}, => e.result.is_none() && !e.completed == true ??

My mistake,  only {result: None, completed: false} => `e.result.is_none() && !e.completed == true`.

Reply:
> There is a tricky point as if itself has the result, the self result is not none because of the edition, so basically completed == is_none(). Here only use completed only maybe ok.

I think only use `e.completed` is OK